packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -1,29 +1,493 @@
 
     before('create users', () => testUsers.create({count: 1})
       .then(([createdUser]) => {
         user = createdUser;
       }));
 
-    beforeEach('create the webex instance', () => {});
+    beforeEach('create the webex instance', () => {
+      webex = new WebexCore({
+        credentials: user.token
+      });
+
+      devices = webex.internal.devices;
+    });
+
+    // This will be implemented in the same pull request that `unregister()` is
+    // implemented.
     afterEach('attempt to unregister the device', () => {});
+
     after('remove users', () => testUsers.remove([user]));
 
-    describe('#canRegister()', () => {});
-    describe('#checkNetworkReachability()', () => {});
+    describe('events', () => {
+      describe('when a meeting is started', () => {
+        beforeEach('setup sinon', () => {
+          devices.resetLogoutTimer = sinon.spy();
+          webex.trigger('meeting started');
+        });
+
+        it('should set \'isInMeeting\' to \'true\'', () => {
+          assert.isTrue(devices.isInMeeting);
+        });
+
+        it('should call \'resetLogoutTimer()\'', () => {
+          assert.called(devices.resetLogoutTimer);
+        });
+      });
+
+      describe('when a meeting has ended', () => {
+        beforeEach('setup sinon', () => {
+          devices.resetLogoutTimer = sinon.spy();
+          devices.isInMeeting = false;
+          webex.trigger('meeting ended');
+        });
+
+        it('should set \'isInMeeting\' to \'false\'', () => {
+          assert.isFalse(devices.isInMeeting);
+        });
+
+        it('should call \'resetLogoutTimer()\'', () => {
+          assert.called(devices.resetLogoutTimer);
+        });
+      });
+    });
+
+    describe('#canRegister()', () => {
+      describe('when wait is truthy', () => {
+        let wait;
+
+        beforeEach('set wait', () => {
+          wait = true;
+        });
+
+        describe('when the `wdm` service is available', () => {
+          let services;
+
+          beforeEach('destructure services plugin and get catalog', () => {
+            services = webex.internal.services;
+
+            return services.waitForCatalog('postauth');
+          });
+
+          it('should return a resolved promise', () => {
+            assert.isDefined(services.get('wdm'));
+
+            return devices.canRegister(wait)
+              .then(() => assert.isTrue(true, 'resolved the promise'));
+          });
+        });
+
+        describe('when the service catalog is not ready', () => {
+          let services;
+
+          beforeEach('setup catalog to be not ready', () => {
+            services = webex.internal.services;
+
+            services.updateServices();
+          });
+
+          it('should wait for the catalog and then resolve', () => {
+            /* eslint-disable-next-line no-underscore-dangle */
+            assert.isTrue(services._getCatalog().status.postauth.collecting);
+
+            return devices.canRegister(wait)
```

Comment:
Using chai-as-promised you can do the following:

```suggestion
            assert. isFulfilled(devices.canRegister(wait));
```

https://github.com/webex/webex-js-sdk/blob/master/packages/node_modules/@webex/test-users/test/integration/spec/index.js#L110